crafting_combinator/script/rc.lua

```diff
@@ -22,6 +22,16 @@ function _M.init_global()
 	global.rc = global.rc or {}
 	global.rc.data = global.rc.data or {}
 	global.rc.ordered = global.rc.ordered or {}
+	local crafting_machines = {}
+	for name,prototype in pairs(game.entity_prototypes) do
+		if prototype.crafting_categories and game.forces[1].recipes[name] then
+			for categorie in pairs(prototype.crafting_categories) do
+				crafting_machines[categorie] = crafting_machines[categorie] or {}
+				table.insert(crafting_machines[categorie],name)
+			end
+		end
+	end
+	global.rc.machines = crafting_machines
```

Comment:
Is there a particular reason this needs to be in global? I'd prefer if it was just a local that gets rebuilt when necessary.